src/Cms/AppPlugins.php

```diff
@@ -885,14 +885,25 @@ protected function pluginsLoader(): array
                 continue;
             }
 
-            $dir   = $root . '/' . $dirname;
-            $entry = $dir . '/index.php';
+            $dir = $root . '/' . $dirname;
 
-            if (is_dir($dir) !== true || is_file($entry) !== true) {
+            if (is_dir($dir) !== true) {
                 continue;
             }
 
-            F::loadOnce($entry);
+            $entry  = $dir . '/index.php';
+            $script = $dir . '/index.js';
+            $styles = $dir . '/index.css';
+
+            if (is_file($entry) === true) {
+                F::loadOnce($entry);
+            } elseif (is_file($script) === true || is_file($styles) === true) {
+                // if no PHP file is present but an index.js or index.css,
+                // register as anonymous plugin (without actual extensions)
+                // to be picked up by the Panel\Document class when
+                // rendering the Panel view
+                static::plugin('plugins/' . $dirname, ['root' => $dir]);
+            }
 
             $loaded[] = $dir;
```

Comment:
Not sure about the implications of this line being run even for empty directories now. Previously this line would only be reached if an `index.php` was in the directory, but now even for empty directories.

And with "not sure" I mean really that. Doesn't mean that we need to change it.

Reply:
What do think about adding `continue` within `else` fallback?

Reply:
Would work as well